js/kraken.js

```diff
@@ -528,6 +530,141 @@ module.exports = class kraken extends Exchange {
+        // https://www.kraken.com/features/api#get-ledgers-info
+        await this.loadMarkets ();
+        let request = {};
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['asset'] = currency['id'];
+        }
+        if (since !== undefined) {
+            request['start'] = parseInt (since / 1000);
+        }
+        let response = await this.privatePostLedgers (this.extend (request, params));
+        // {  error: [],
+        //   result: { ledger: { 'LPUAIB-TS774-UKHP7X': {   refid: "A2B4HBV-L4MDIE-JU4N3N",
+        //                                                   time:  1520103488.314,
+        //                                                   type: "withdrawal",
+        //                                                 aclass: "currency",
+        //                                                  asset: "XETH",
+        //                                                 amount: "-0.2805800000",
+        //                                                    fee: "0.0050000000",
+        //                                                balance: "0.0000051000"           },
+        let ledger = response['result']['ledger'];
+        let keys = Object.keys (ledger);
+        let data = [];
+        for (let i = 0; i < keys.length; i++) {
+            let key = keys[i];
+            let value = ledger[key];
+            value['id'] = key;
+            data.push (value);
+        }
+        return this.parseLedger (data, currency, since, limit);
+    }
+
+    async fetchLedgerItemsByIds (ids, code = undefined, params = {}) {
+        // https://www.kraken.com/features/api#query-ledgers
+        await this.loadMarkets ();
+        ids = ids.join (',');
+        let request = this.extend ({
+            'id': ids,
+        }, params);
+        let response = await this.privatePostQueryLedgers (request);
+        // {  error: [],
+        //   result: { 'LPUAIB-TS774-UKHP7X': {   refid: "A2B4HBV-L4MDIE-JU4N3N",
+        //                                         time:  1520103488.314,
+        //                                         type: "withdrawal",
+        //                                       aclass: "currency",
+        //                                        asset: "XETH",
+        //                                       amount: "-0.2805800000",
+        //                                          fee: "0.0050000000",
+        //                                      balance: "0.0000051000"           } } }
+        let ledger = response['result'];
+        let keys = Object.keys (ledger);
+        let data = [];
+        for (let i = 0; i < keys.length; i++) {
+            let key = keys[i];
+            let value = ledger[key];
+            value['id'] = key;
+            data.push (value);
+        }
+        return this.parseLedger (data);
```

Comment:
I mean, this particular line is missing the filter.

Reply:
> I mean, this particular line is missing the filter.

It does so because we have neither `since` nor `limit` nor `currency` within this method. You are requesting particular ledger items specifying their ids with this method – so i don't see how can we use `currency`, `since` and `limit` here.

Reply:
> Yeah, right... I don't really like this method, though.

I would say that in case of Kraken this method is quite useless since it does not provide any additional information about ledger item on top of `fetchLedger` response. However some other exchange may provide it, and that's why i think it is a good idea to have unified `fetchLedgerItem` method. It turns out that Kraken allows you to fetch multiple items at once and i don's see a reason why we should restrict this functionality for our users (even though we think it is useless).